apps/withdraw-pool/src/utils/decodeHookCalldata.ts

```diff
@@ -27,5 +34,35 @@ export function findPoolIdOnCallData(string: `0x${string}`) {
 
   const lastCallIndex = calls.length - 1;
   const call = calls[lastCallIndex];
-  return call.target.toLowerCase();
+
+  const decodedCoWAmmCallData = decodeFunctionData({
+    abi: cowAmmAbi,
+    data: call.callData as `0x${string}`,
+  });
+
+  const poolAddress = call.target;
+
+  if (decodedCoWAmmCallData.functionName !== "exitPool") {
+    return;
+  }
+
+  if (!decodedCoWAmmCallData.args?.length) {
+    return;
+  }
+
+  const bptAmountToExit = decodedCoWAmmCallData.args[0] as bigint;
+
+  const userPoolBalance = (await publicClient.readContract({
+    address: poolAddress as `0x${string}`,
+    abi: cowAmmAbi,
+    functionName: "balanceOf",
+    args: [user],
+  })) as bigint;
+
+  const withdrawPct = getPctFromValue(bptAmountToExit, userPoolBalance);
```

Comment:
`#would` rename this to help understand we're using the same units

```suggestion
  const withdrawPct = getPctFromValue(bptAmountToExit, bptUserPoolBalance);
```